src/olympia/shelves/tests/test_forms.py

```diff
@@ -116,3 +117,32 @@ def test_clean_returns_empty(self):
             form.clean()
         assert exc.exception.message == (
             u'Check criteria parameters - e.g., "type"')
+
+
+class TestShelfManagementForm(TestCase):
+    def setUp(self):
+        self.shelf = Shelf.objects.create(
+            title='Recommended extensions',
+            endpoint='search',
+            criteria='?recommended=true&sort=random&type=extension',
+            footer_text='See more recommended extensions',
+            footer_pathname='/this/is/the/pathname')
+
+    def test_clean(self):
+        form = ShelfManagementForm({
+            'enabled': True,
+            'position': 1,
+            'shelf': self.shelf.pk})
+        assert form.is_valid(), form.errors
+        assert form.cleaned_data['enabled']
+        assert form.cleaned_data['position'] == 1
+
+    def test_clean_position_required_if_enabled(self):
+        form = ShelfManagementForm({
+            'enabled': True,
+            'shelf': self.shelf.pk})
+        assert not form.is_valid(), form.errors
+        with self.assertRaises(ValidationError) as exc:
+            form.clean()
+        assert exc.exception.message == (
+            u'Position field is required to enable shelf.')
```

Comment:
No need for `u'...'` strings - we're way past python2 compatibility 😄  (they only still exist in the codebase because it will be a huge patch to remove them all)